helix-view/src/gutter.rs

```diff
@@ -265,30 +265,50 @@ pub fn breakpoints<'doc>(
                 .iter()
                 .find(|breakpoint| breakpoint.line == line)?;
 
-            let mut style = if breakpoint.condition.is_some() && breakpoint.log_message.is_some() {
+            let style = if breakpoint.condition.is_some() && breakpoint.log_message.is_some() {
                 error.underline_style(UnderlineStyle::Line)
             } else if breakpoint.condition.is_some() {
                 error
             } else if breakpoint.log_message.is_some() {
                 info
             } else {
-                warning
+                breakpoint_style
             };
 
-            if !breakpoint.verified {
-                // Faded colors
-                style = if let Some(Color::Rgb(r, g, b)) = style.fg {
-                    style.fg(Color::Rgb(
-                        ((r as f32) * 0.4).floor() as u8,
-                        ((g as f32) * 0.4).floor() as u8,
-                        ((b as f32) * 0.4).floor() as u8,
-                    ))
-                } else {
-                    style.fg(Color::Gray)
-                }
-            };
+            let sym = if breakpoint.verified { "●" } else { "◯" };
+            write!(out, "{}", sym).unwrap();
+            Some(style)
+        },
+    )
+}
+
+fn execution_pause_indicator<'doc>(
+    editor: &'doc Editor,
+    doc: &'doc Document,
+    theme: &Theme,
+    is_focused: bool,
+) -> GutterFn<'doc> {
+    let style = theme.get("ui.debug.active");
+
+    Box::new(
+        move |line: usize, _selected: bool, first_visual_line: bool, out: &mut String| {
+            if !first_visual_line || !is_focused {
+                return None;
+            }
+
+            let current_stack_frame = editor.current_stack_frame()?;
+            if line != current_stack_frame.line - 1
+                || doc.path().is_none()
+                || current_stack_frame
+                    .source
+                    .as_ref()
+                    .and_then(|source| source.path.as_ref())
+                    != doc.path()
```

Comment:
I would also extract this path check outside the closure (`let is_current = ...`), then return at the top of the closure if no match.

Reply:
That way the only thing you need to save outside the closure really is `let stack_fraame_line = current_stack_frame.line - 1`